File: app/src/main/java/com/example/repaircalculator/adapter/ProjectAdapter.java
package com.example.repaircalculator.adapter;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.repaircalculator.data.entity.Project;
import com.example.repaircalculator.databinding.ProjectElementBinding;
import com.example.repaircalculator.viewHolder.ProjectViewHolder;

import java.util.Collections;
import java.util.List;

public class ProjectAdapter extends RecyclerView.Adapter<ProjectViewHolder> {

    private List<Project> projects = Collections.emptyList();

    @NonNull
    @Override
    public ProjectViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflate = LayoutInflater.from(parent.getContext());
        ProjectElementBinding binding = ProjectElementBinding.inflate(inflate, parent, false);
        return new ProjectViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ProjectViewHolder holder, int position) {
        holder.bind(projects.get(position));
    }

    @Override
    public int getItemCount() {
        return projects.size();
    }

    public void setProjects(List<Project> projects) {
        this.projects = projects;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/example/repaircalculator/adapter/ProjectAdapter.kt
package com.example.repaircalculator.adapter

import androidx.recyclerview.widget.RecyclerView
import com.example.repaircalculator.viewHolder.ProjectViewHolder
import com.example.repaircalculator.data.entity.Project
import android.view.ViewGroup
import android.view.LayoutInflater
import com.example.repaircalculator.databinding.ProjectElementBinding

class ProjectAdapter : RecyclerView.Adapter<ProjectViewHolder>() {

    private var projects = emptyList<Project>()
    var categoryCallback: ((Project) -> Unit?)? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ProjectViewHolder {
        val inflate = LayoutInflater.from(parent.context)
        val binding = ProjectElementBinding.inflate(inflate, parent, false)
        return ProjectViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ProjectViewHolder, position: Int) {
        holder.bind(projects[position], categoryCallback)
    }

    override fun getItemCount(): Int {
        return projects.size
    }

    fun setProjects(projects: List<Project>) {
        this.projects = projects
        notifyDataSetChanged()
    }
}